src/Http/Http/src/Timeouts/RequestTimeoutsMiddleware.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http.Timeouts;
+
+internal sealed class RequestTimeoutsMiddleware
+{
+    private readonly RequestDelegate _next;
+    private readonly IOptionsMonitor<RequestTimeoutOptions> _options;
+
+    public RequestTimeoutsMiddleware(RequestDelegate next, IOptionsMonitor<RequestTimeoutOptions> options)
+    {
+        _next = next;
+        _options = options;
+    }
+
+    public Task Invoke(HttpContext context)
+    {
+        var endpoint = context.GetEndpoint();
+        var timeoutMetadata = endpoint?.Metadata.GetRequiredMetadata<RequestTimeoutAttribute>();
```

Comment:
I think the metadata can be optional (not required) also an endpoint doesn't need to have all of them right?

Reply:
```suggestion
        var timeoutMetadata = endpoint?.Metadata.GetMetadata<RequestTimeoutAttribute>();
```